onnxruntime/core/providers/tensorrt/tensorrt_execution_provider_info.cc

```diff
@@ -360,6 +360,9 @@ void TensorrtExecutionProviderInfo::UpdateProviderOptions(void* provider_options
   trt_provider_options_v2.trt_engine_hw_compatible = internal_options.engine_hw_compatible;
   trt_provider_options_v2.trt_onnx_bytestream = internal_options.onnx_bytestream;
   trt_provider_options_v2.trt_onnx_bytestream_size = internal_options.onnx_bytestream_size;
-  trt_provider_options_v2.trt_op_types_to_exclude = copy_string_if_needed(internal_options.op_types_to_exclude);
+  if (options.find("trt_op_types_to_exclude") != options.end()) {
+    trt_provider_options_v2.trt_op_types_to_exclude = copy_string_if_needed(internal_options.op_types_to_exclude);
```

Comment:
copy_string_if_needed() doesn't always dynamically allocate memory right? only if string_copy is true?
this is_dynamic_allocation check looks ugly.

Reply:
Yes, i can add a check that only if string_copy is true, we set is_dynamic_allocation to true.

yeah, agree the is_dynamic_allocation is ugly. But i can't think of a better solution without having a flag.
The reason is options_v2.trt_op_types_to_exclude which is a const char* can be static allocation or dynamic allocation.

- static allocation - When instantiate an OrtTensorRTProviderOptionsV2 instance, the [trt_op_types_to_exclude](https://github.com/microsoft/onnxruntime/blob/765cb7ff80e6595f7677ed8d4f7c0b69d2183240/include/onnxruntime/core/providers/tensorrt/tensorrt_provider_options.h#L90) is static allocation.
- dynamic allocation - When user calls OrtApis::ReleaseTensorRTProviderOptions and specify other value to trt_op_types_to_exclude, then it's dynamic allocation.

So, once ReleaseTensorRTProviderOptions is being called, it's hard to tell that trt_op_types_to_exclude is static allocation or dynamic allocation. We can't delete a static allocation buffer.

Reply:
It's an internal file. Why don't we just use a std::string? Then there will be no need to manually delete[] buffers.

Reply:
We later discussed we will close this PR due to ugly handling of the buffers.